File: bigdime-data-handlers/src/main/java/io/bigdime/handler/hive/JobClientFactory.java
package io.bigdime.handler.hive;

import java.io.IOException;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.mapred.JobClient;
import org.springframework.stereotype.Component;

@Component
public class JobClientFactory {
	public JobClient createJobClient(Configuration conf) throws IOException {
		return new JobClient(conf);
	}
}

File: bigdime-libraries/src/main/scala/io/bigdime/libs/hive/job/JobClientFactory.scala
package io.bigdime.libs.hive.job

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.mapred.JobClient
import org.springframework.stereotype.Component

import java.io.IOException

@Component
class JobClientFactory {
  @throws[IOException]
  def createJobClient(conf: Configuration): JobClient = {
    return new JobClient(conf)
  }
}
